### PR TITLE
Fix dump subcommand in 8.x

### PR DESCRIPTION
**Pull request by jsoriano**
`dump` command failed to discover index templates in 8.x because `managed_by` meta
value was changed from `ingest-manager` to `fleet`, possibly in https://github.com/elastic/kibana/pull/119380.

Add support for 8.x by searching also index templates with `managed_by: fleet`, and adding
a test case.

### CODE REVIEW

File: internal/dump/installedobjects_test.go [hunk 1]
```diff
@@ -15,30 +16,87 @@ import (
...
-func TestInstalledObjectsDumpAll(t *testing.T) {
-	client := estest.ElasticsearchClient(t, "./testdata/elasticsearch-mock-dump-apache")
-	outputDir := t.TempDir()
-	dumper := NewInstalledObjectsDumper(client, "apache")
+func TestDumpInstalledObjects(t *testing.T) {
+	// Files for each suite are recorded automatically on first test run.
+	// To add a new suite:
+	// - Configure it here.
+	// - Install the package in a running stack.
+	// - Configure environment variables for this stack (eval "$(elastic-package stack shellinit)").
```
Comment (mtojek) [inline internal/dump/installedobjects_test.go:31]: I might have missed something introduced in the previous PR, but does it require the Elasticsearch to be up or does it still operate on mocks?

File: internal/dump/installedobjects_test.go [hunk 2]
```diff
@@ -15,30 +16,87 @@ import (
...
+			DumpDir:     "./testdata/elasticsearch-8-apache-dump-all",
+		},
+	}
+
+	for _, s := range suites {
+		suite.Run(t, s)
+	}
+}
+
+type InstalledObjectsDumpSuite struct {
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply internal/dump/installedobjects_test.go]: Not needed, changed to lowercase. In any case as this is a test file it wouldn't have been exposed out of this package.

File: internal/dump/installedobjects_test.go [hunk 3]
```diff
@@ -15,30 +16,87 @@ import (
...
+		s.Require().Greater(n, 0)
+	} else {
+		s.Require().NoError(err)
+	}
+}
+
+func (s *InstalledObjectsDumpSuite) TestDumpAll() {
+	client := estest.ElasticsearchClient(s.T(), s.RecordDir)
+
+	outputDir := s.T().TempDir()
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply internal/dump/installedobjects_test.go:83]: Directories created with [`TempDir()`](https://pkg.go.dev/testing#B.TempDir) are automatically removed by `Cleanup`.

**Review by mtojek — APPROVED** [review]

**Pull request merged**